swap_meet/vendor.py

```diff
@@ -1,2 +1,67 @@
+from .item import Item
+
 class Vendor:
-    pass
+    def __init__(self, inventory = None):
+        if inventory:
+            self.inventory = inventory
+        else:
+            self.inventory = []
+    
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+    
+    def remove(self, item):
+        if item in self.inventory:
+            self.inventory.remove(item)
+            return item
+        else:
+            return False
+    
+    def get_by_category(self, category):
+        category_items = []
+        for item in self.inventory:
+            if item.category == category:
+                category_items.append(item)
+        
+        return category_items
+    
+    def swap_items(self, friend, my_item, their_item):
+        if my_item in self.inventory and their_item in friend.inventory:
+            self.remove(my_item)
+            friend.remove(their_item)
+            friend.add(my_item)
+            self.add(their_item)
+            return True
+        else:
+            return False
+
+    def swap_first_item(self, friend):
+        if self.inventory and friend.inventory:
+            self.swap_items(friend, self.inventory[0], friend.inventory[0])
+            return True
+
+    def get_best_by_category(self, category):
```

Comment:
Great use of `get_by_category` as a helper method and using a flag for `best_item`.